subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc

```diff
@@ -38,6 +38,39 @@ The following source tree contains Java and Kotlin source files. Java source fil
             └── Utils.kt
 ----
 
+[[sec:separate_test_type_source_files]]
+=== Separate source files per test type
+
+It's very common that a project defines and executes different types of tests e.g. unit tests, integration tests, functional tests or smoke tests. Optimally, the test source code for each test type should be stored in dedicated source directories. Separated test source code has a positive impact on maintainability and separation of concerns.
+
+The following source tree demonstrates how to separate unit from integration tests in a Java-based project.
+
+----
+.
+├── build.gradle
+├── gradle
+│   └── integration-test.gradle
+├── settings.gradle
+└── src
+    ├── integTest
+    │   └── java
+    │       └── DefaultFileReaderIntegrationTest.java
+    ├── main
+    │   └── java
+    │       ├── DefaultFileReader.java
+    │       ├── FileReader.java
+    │       └── StringUtils.java
+    └── test
+        └── java
+            └── StringUtilsTest.java
+----
+
+Gradle models source code directories with the help of the <<sec:working_with_java_source_sets,source set concept>>. By pointing an instance of a source set to one or many source code directories, Gradle will automatically create a corresponding compilation task out-of-the-box.
+
+Source sets are only responsible for compiling source code, but do not deal with executing the byte code. For the purpose of test execution, a corresponding task of type api:org.gradle.api.tasks.testing.Test[] needs to be established.
+
+The user guide won't go into detail on the implementation details. You can find a sample project in the `samples` directory of the standard `-all` Gradle distribution.
```

Comment:
That's what I mention in the description of the issue:

> I left of a lot of code examples on purpose as it would blow up the chapter. Instead I'd like to point directly to the code in the -all distribution. Right now there's nothing built-in to do that well.

We'd have to build a new documentation feature to be able to point to a concrete path in the distribution. Alternatively, we thought about linking to some online source rendering all sample code.

The other option would simply be to embed some code samples though I think it would blow up the page unnecessarily. I think of this page as a collection of recipes.